syntax = "proto3";

package training;

service MetricsService {
  rpc ReportMetrics(MetricData) returns (Ack);
  rpc GetAllMetrics(Empty) returns (MetricsList);
}

message Empty {}

message Ack {
  bool success = 1;
  string message = 2;
}

message MetricData {
  string worker_id = 1;
  int32 iteration = 2;
  float loss = 3;
  float accuracy = 4;
  string timestamp = 5;
}

message MetricsList {
  repeated MetricData metrics = 1;
}